src/library/scala/collection/Iterable.scala

```diff
@@ -551,6 +551,31 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
       else View.Empty
     }
 
+  /** A pair of ${coll}s, first, consisting of the values that are produced by `f` applied to this $coll elements contained in [[scala.util.Left]] and, second,
+    *  all the values contained in [[scala.util.Right]].
+    *
+    *  Example:
+    *  {{{
+    *    val xs = $Coll(1, "one", 2, "two", 3, "three") mapSplit {
+    *     case i: Int => Left(i)
+    *     case s: String => Right(s)
+    *    }
+    *    // xs == ($Coll(1, 2, 3),
+    *    //        $Coll(one, two, three))
+    *  }}}
+    *
+    *  @tparam A1         element type of the first resulting collection
+    *  @tparam A2         element type of the second resulting collection
+    *  @param f          split function that map the element of the $coll into an [[scala.util.Either]][A1, A2]
+    *
+    *  @return           a pair of ${coll}s, first, consisting of the values that are produced by `f` applied to this $coll
+    *               elements contained in [[scala.util.Left]] and, second, all the values contained in [[scala.util.Right]].
+    */
+  def mapSplit[A1, A2](f: A => Either[A1, A2]): (CC[A1], CC[A2]) = {
```

Comment:
I agree with the comments on the ticket about the name. I think we should name it `partitionWith`. The kind of splitting by predicate that it does is called "partition" in other cases ("split" usually refers to splitting by index) and there's precedence for `partitionWith` in other libraries.